plugins/rest/aws.go

```diff
@@ -318,6 +318,150 @@ func (cs *awsMetadataCredentialService) credentials(ctx context.Context) (aws.Cr
 	return cs.creds, nil
 }
 
+// awsCustomIdentityCredentialService represents an STS credential service that uses active IAM credentials
+// to obtain temporary security credentials generated by AWS STS for allowing existing IAM users access to AWS resources
+// that they don't already have access to.
+type awsCustomIdentityCredentialService struct {
+	RoleArn          string
+	AccessKey        string
+	SecretKey        string
+	SessionToken     string
```

Comment:
Are these attributes public because they're expected to be read/written somewhere (as e.g. json, not from env vars, as in `populateFromEnv()`), or could they just as well be private?

Reply:
They're currently expected to be written via env vars. I've tried to stay consistent with what we have done for the other credential providers.

Reply:
@ashutosh-narkar Will reading from custom profile reds file also be supported here?